src/main/java/org/broadinstitute/hellbender/transformers/DUSTReadTransformer.java

```diff
@@ -0,0 +1,230 @@
+                    cv[s]--;
+                    rv -= cv[s];
+                    L--;
+                } while (s != t);
+            }
+            //End: SHIFT_WINDOW
+
+            if (rw*10 > L*T) {
+                //Begin: FIND_PERFECT
+
+                short[] c = cv.clone();
+                int r = rv;
+                ListIterator<PStruct> iter = P.listIterator();
+                PStruct perf = iter.hasNext() ? iter.next() : null;
+                float max_score = 0;
+                for (int i = w.size() - L - 1; i >= 0; i--) {
+                    t = w.get(i);
+                    r += c[t];
+                    c[t]++;
+                    float new_score = ((float)r)/(w.size() - i - 1);
+                    if (new_score*10 > T) {
+                        while (perf != null && perf.start >= i + wstart) {
+                            max_score = Math.max(max_score,perf.score);
+                            perf = iter.hasNext() ? iter.next() : null;
+                        }
+                        if (new_score >= max_score) {
+                            max_score = new_score;
+                            PStruct new_perf = new PStruct();
+                            new_perf.start = i + wstart;
+                            new_perf.finish = w.size() + 1 + wstart;
+                            new_perf.score = new_score;
+                            if (iter.hasPrevious()) {
+                                iter.previous();
+                            } else {
+                                iter = P.listIterator();
+                            }
+                            iter.add(new_perf);
+                            if (iter.hasNext()) {iter.next();}
+                        }
+                    }
+                }
+
+                //End: FIND_PERFECT
+            }
+        }
+
+        wstart = Math.max(0,read.getLength() - W + 1);
+
+        PStruct p1;
+        while (!P.isEmpty()) {
+
+            //Begin: SAVE_MASKED_REGIONS (TODO: copied code from above)
+            p1 = P.getLast();
+            if (p1.start < wstart) {
+                int l = res.size();
+                if (l > 0) {
+                    Tuple<Integer,Integer> rt = res.get(l - 1);
+                    if (p1.start < rt.b + 1) {
+                        Tuple<Integer,Integer> new_rt = new Tuple<>(rt.a,Math.max(rt.b,p1.finish));
+                        res.set(l - 1,new_rt);
+                    } else {
+                        res.add(new Tuple<Integer,Integer>(p1.start,p1.finish));
+                    }
+                } else {
+                    res.add(new Tuple<Integer,Integer>(p1.start,p1.finish));
+                }
+                while (p1 != null && p1.start < wstart) {
+                    P.removeLast();
+                    p1 = P.isEmpty() ? null : P.getLast();
+                }
+            }
+            //End: SAVE_MASKED_REGIONS
+
+            wstart++;
+        }
+
```

Comment:
Could you extract all the code up to here as a separate method that returns `res`?  It would be useful in other contexts to be able to find out what the DUST-masked intervals are without having to transform the read.